v3-sdk/modifying-position/src/example/Example.tsx

```diff
@@ -0,0 +1,386 @@
+import React, { useCallback, useEffect, useState } from 'react'
+import './Example.css'
+import { ethers } from 'ethers'
+import {
+  Pool,
+  computePoolAddress,
+  Position,
+  nearestUsableTick,
+  NonfungiblePositionManager,
+} from '@uniswap/v3-sdk'
+import { Percent, CurrencyAmount } from '@uniswap/sdk-core'
+import { Environment, CurrentConfig } from '../config'
+import { getCurrencyBalance } from '../libs/balance'
+import { getPositionIds, getTokenTransferApprovals } from '../libs/positions'
+import {
+  connectBrowserExtensionWallet,
+  getProvider,
+  TransactionState,
+  sendTransaction,
+  getWalletAddress,
+} from '../libs/providers'
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json'
+import {
+  POOL_FACTORY_CONTRACT_ADDRESS,
+  MAX_FEE_PER_GAS,
+  MAX_PRIORITY_FEE_PER_GAS,
+  NONFUNGIBLE_POSITION_MANAGER_CONTRACT_ADDRESS,
+  DAI_TOKEN,
+  USDC_TOKEN,
+} from '../libs/constants'
+import { fromReadableAmount } from '../libs/conversion'
+
+interface PoolInfo {
+  token0: string
+  token1: string
+  fee: number
+  tickSpacing: number
+  sqrtPriceX96: ethers.BigNumber
+  liquidity: ethers.BigNumber
+  tick: number
+}
+
+const getPoolInfo = async (): Promise<PoolInfo> => {
+  const provider = getProvider()
+  if (!provider) {
+    throw new Error('No provider')
+  }
+
+  const currentPoolAddress = computePoolAddress({
+    factoryAddress: POOL_FACTORY_CONTRACT_ADDRESS,
+    tokenA: CurrentConfig.tokens.token0,
+    tokenB: CurrentConfig.tokens.token1,
+    fee: CurrentConfig.tokens.poolFee,
+  })
+
+  const poolContract = new ethers.Contract(
+    currentPoolAddress,
+    IUniswapV3PoolABI.abi,
+    provider
+  )
+
+  const [token0, token1, fee, tickSpacing, liquidity, slot0] =
+    await Promise.all([
+      poolContract.token0(),
+      poolContract.token1(),
+      poolContract.fee(),
+      poolContract.tickSpacing(),
+      poolContract.liquidity(),
+      poolContract.slot0(),
+    ])
+
+  return {
+    token0,
+    token1,
+    fee,
+    tickSpacing,
+    liquidity,
+    sqrtPriceX96: slot0[0],
+    tick: slot0[1],
+  }
+}
+
+const getPosition = async (
+  percentageToIncrease?: number
+): Promise<Position> => {
+  // get pool info
+  const poolInfo = await getPoolInfo()
+
+  // construct pool instance
+  const USDC_DAI_POOL = new Pool(
```

Comment:
Do we want to break this into the `positions.ts` file now or in a followup when we do this for others?

Reply:
I can address this now :)